point_pattern.py

```diff
@@ -0,0 +1,87 @@
+from .point import Point
+from . import analytics
+import random
+import numpy as np
+
+
+class PointPattern(object):
+    def __init__(self):
+        self.points = []
+
+    def add_point(self, point):
+        self.points.append(point)
+
+    def remove_point(self, index):
+        try:
+            del(self.points[index])
+        except IndexError:
+            print('Index {} not in list'.format(index))
+
+    def average_nearest_neighbor_distance(self, mark=None):
+        return analytics.average_nearest_neighbor_distance(self.points, mark)
+
+    def count_coincident(self):
+        """
+        Returns the number of coincident points.
+        If two points are at the same spatial location,
+        that counts as two coincident points. Three coincident points
+        means three points at the same location.
+        """
+        to_return = 0
+        handled_indices = []
+        for i, point_a in enumerate(self.points):
+            for j, point_b in enumerate(self.points):
+                if j in handled_indices:
+                    continue
+                if i == j:
+                    continue
+                if point_a.is_coincident(point_b):
+                    to_return += 1
+                    handled_indices.append(j)
+        return to_return
+
+    def list_marks(self):
+        marks = []
+        for point in self.points:
+            if 'color' in point.mark and point.mark['color'] not in marks:
+                marks.append(point.mark['color'])
```

Comment:
Another way to do this is to utilize a `set`.  https://docs.python.org/3.5/library/stdtypes.html?highlight=set#set  This can be more efficient than the `if` statement.